src/hooks.rs

```diff
@@ -10,6 +10,16 @@ use crate::hints::MOCK_CALL_KEY;
 
 pub const HOOKS_VAR_NAME: &str = "hooks";
 
+/// Called before an instruction is executed by the virtual machine (VM).
+/// 
+/// This function is called before the VM will execute an instruction.
+/// The given `&mut VirtualMachine` allow to modify VM state
+/// according to given `&mut ExecutionScopes`.
+/// 
+/// Returns nothing with means that everything is fine.
```

Comment:
> Returns nothing with means that everything is fine.

-> 

> When no error is encountered, returns an empty success value.
> Otherwise, returns a VirtualMachineError